### PR TITLE
Refactor extracting

### PR DESCRIPTION
**Pull request by cdrage**
This commit refactors extracting and separates the function into two
calls. Extracting the files from the Docker container and extracting the
Nulecule-specific files.

cc @dustymabe

### CODE REVIEW

**Comment by cdrage** [comment]
#dotests

**Comment by dustymabe** [comment]
The diff would be alot cleaner if you left the `is_image_present(self, image)` function at the bottom of the file.

**Comment by dustymabe** [comment]
Do you mind making the removal of the `There has been some inconsistent behavior` a separate commit? I image the functionality has stabilized now, would be nice to test and then state that in the commit msg.

**Comment by cdrage** [comment]
Much cleaner :) Thanks for reviewing @dustymabe

File: atomicapp/nulecule/container.py [hunk 1]
```diff
@@ -93,41 +93,67 @@ def pull(self, image, update=False):
...
+    def extract_files(self, image, source, dest):
         """
-        Extracts content from a directory in a Docker image to specified
+        Extracts a directory/file in a Docker image to a specified
         destination.
 
         Args:
             image (str): Docker image name
             source (str): Source directory in Docker image to copy from
             dest (str): Path to destination directory on host
```
Comment (dustymabe) [inline atomicapp/nulecule/container.py]: Might want to update these comments to say file or directory? - or you can do that later if you want in the "index" PR

**Comment by dustymabe** [comment]
LGTM

**Comment by cdrage** [comment]
Thanks :) All tests pass (double-checked my side running unit + functional tests too)

**Pull request merged**